style(notification-badge): Extend badge props + Refactor style

housseindjirdeh opened the pull request:
Update notifications icon badge:

1. Made it blue and text more prominent
2. Moved the `absolute` style logic to outside the badge component. If we plan on using this component elsewhere it shouldn't have absolute styling within
3. Extended props to include color and background color

CC @lex111 

![screen shot 2017-09-30 at 11 29 28 pm](https://user-images.githubusercontent.com/12476932/31051546-f66eaadc-a638-11e7-81af-4c4a4b89f383.png)

lex111 commented:
@housseindjirdeh could set `-1` for `top` and `right`, and also return `borderColor` and `borderWidth`? So it looks better to the badge in the tabbar.

lex111 commented on src/components/notification-icon.component.js:
```diff
@@ -1,19 +1,33 @@
+const styles = StyleSheet.create({
+  icon: {
+    justifyContent: 'center',
+    alignItems: 'center',
```
Need to add `flex: 1`.

chinesedfan commented:
@housseindjirdeh I have never seen any blue notifications. I prefer to keep it as red. :)

lex111 commented on src/components/notification-icon.component.js:
```diff
@@ -1,19 +1,33 @@
+import { colors } from 'config';
+
+const styles = StyleSheet.create({
+  icon: {
```
This is still my fault, but we will not have alignment, without `flex: 1`, so it can move the styles to the `badgeContainer`? Though all the same the vertical alignment not absolutely on the center seems ..

housseindjirdeh replied:
You're right, adding `flex: 1` messes up positioning entirely so removed this style completely and it looks fine without :+1:

housseindjirdeh commented:
Cheers thank you @chinesedfan @lex111. Thought blue would better match GitHub's current badge color but probably a bit head of myself :joy: 

![image](https://user-images.githubusercontent.com/12476932/31056555-7c293ee0-a6a1-11e7-90e2-3624947ad22d.png)

Reverted back to dark red and added the border back.

![image](https://user-images.githubusercontent.com/12476932/31056563-9ef48a2e-a6a1-11e7-8dd5-b7f8c701881b.png)

Also used `-1` positioning like you mentioned @lex111 :dancer:

lex111 commented:
@housseindjirdeh thanks! Strange border :confused: 
![image](https://user-images.githubusercontent.com/4408379/31056866-8bae0f54-a6e1-11e7-9cd0-7d6a8829bfcc.png)

housseindjirdeh commented:
Left it to the same border settings you had previously @lex111 so not sure why if it looks weird to you  :(

Zoomed in side by side comparison of this PR update and what's currently on master:

![image](https://user-images.githubusercontent.com/12476932/31056931-4562b448-a6a8-11e7-89e0-31fd09c1ea52.png)

lex111 commented:
On Android, simply the color of the border is uniform:

![image](https://user-images.githubusercontent.com/4408379/31056989-d828d09c-a6e3-11e7-9904-b37229c0ede1.png)

housseindjirdeh commented:
Oh does it look more uniform on Android? Not sure why :thinking: 

Again we're zooming in for this, I don't think it looks weird normally on view:

![image](https://user-images.githubusercontent.com/12476932/31057240-8ea316ac-a6ad-11e7-910f-5b8c1a9aaa8b.png)

lex111 commented:
@housseindjirdeh I meant that iOS can see a red background around the border, compare the screenshots, I marked with an arrow. I do not understand why this is so, or is it in dev mode only?

housseindjirdeh commented:
Really not sure, I'm hoping this shows on simulator and not on the actual device. Will test it on my phone once we have it merged in. It is weird though I agree :thinking:
